app/assets/javascripts/client/directives/assessments/assessmentLinks.js

```diff
@@ -93,13 +100,21 @@ PDRClient.directive('assessmentLinks', [
             };
 
             $scope.submitAccessRequest = function(roles) {
+              $scope.performAccessRequest(roles).then(function() {
+                $scope.modal.dismiss('cancel');
+              })
+            };
+
+            $scope.performAccessRequest = function(role) {
```

Comment:
Okay - this was a bit more complicated than I wanted to read, but I think it makes sense.

You've removed the `$scope.modal.dismiss` operation from within the `$promise` below so that you can pick and choose if you need the modal to be dismissed or not.  In one context it does need to be dismissed, but in the other it doesn't.

Is that what you had in mind for this particular code block?